fix: Revised PDP - Action Menus (M2-7828)

farmerpaul opened the pull request:
<!-- Use this template as a guide to describe your pull request, and adjust as necessary. -->
<!-- Include information that helps your peers review your updates and understand this    -->
<!-- repository's history of changes over time.                                           -->

<!-- Delete any options that are not relevant -->

- [ ] ~~Tests for the changes have been added~~ **(to be completed in separate ticket)**
- [ ] Delivered the `fix` or `feature` branches into `develop` or `release` branches via `Squash and Merge` (to keep clean history)

### 📝 Description

<!-- Contributions are welcome! If there is a corresponding      -->
<!-- JIRA ticket, link to it by replacing `#` with ticket number -->

🔗 [Jira Ticket M2-7828](https://mindlogger.atlassian.net/browse/M2-7828)

This PR is mainly here to provide QA with a preview environment to verify the AC of M2-7828, which is about displaying the correct menu items, states, and behaviours for each activity/flow's action menu (⋯). However, the vast majority of the implementation was either already in `develop` or added in currently active PRs.

The only actual code changes done here were:
- Making sure the pre-population alerts in the Assign Activity panel are accurate, based on the nature of the participant(s) being assigned
- Disabling hidden activities/flows in the Assign Activity panel, to align with the disabled state of the corresponding menu item in hidden activities/flows' ⋯ menu

### 📸 Screenshots

N/A

### 🪤 Peer Testing

I highly recommend leaving it to the QA team to go through the AC for this ticket and ensure behaviour of the app aligns, though feel free to run some test cases yourself.

To test the actual code changes introduced by this branch:

**Preconditions:**
- Have an applet with at least one Full Account participant, and a Limited Account participant on the Participants tab.
- Have a manually assigned activity or flow that for which there the Full Account is assigned as the respondent, one with themself as the subject (self-report) and another with the Limited Account as the subject (multi-informant).

**Applet > Participants tab:**
1. Click the ⋯ menu beside the Full Account in the list of Participants and choose the "Assign Activity" menu item.
2. Ensure the banner `1 Participant was added into the table, select an Activity to continue.` is shown in the Assign Activity panel.
3. Click the ⋯ menu beside the admin participant (Team Member) in the list of Participants and choose the "Assign Activity" menu item.
4. Ensure the banner `1 Respondent was added into the table, select an Activity and Subject to continue.` is shown in the Assign Activity panel.

**Participant Details Page (PDP):**
1. Navigate to that participant's PDP and the "By Participant" tab.
4. For any of the activities/flows, click its ⋯ menu (from the unexpanded view) and then choose the "Assign Activity" or "Assign Flow" menu item.
5. Ensure the banner `The Respondent & Activity have been auto-filled, select a Subject to continue.` is shown in the Assign Activity panel.
6. Close the Assign Activity panel.
7. Expand one of the activities/flows, click the ⋯ menu on the subject row that is the self-report, and then choose the "Assign Activity" or "Assign Flow" menu item.
8.  Ensure the banner `The Participant & Activity have been auto-filled, click ‘Next’ to continue.` is shown in the Assign Activity panel.
9. Close the Assign Activity panel.
10. From the same expanded view, click the ⋯ menu on a subject row that a different participant (multi-informant), and then choose the "Assign Activity" or "Assign Flow" menu item.
11. Ensure the banner `The Participants & Activity have been auto-filled, click ‘Next’ to continue.` is shown in the Assign Activity panel.

farmerpaul committed:
fix: update assign prepopulation alert

Better handle edge cases of prepopulating the `ActivityAssignDrawer`
when at least a respondent has been preselected.

JustinNusca reviewed (approved):
Focused on the last commit and the testing notes as described in the PR, like you said — the correct banners are appearing in the appropriate conditions, and I don't have any further comments on the code. LGTM! 👍

LashaunnaS reviewed (approved):
As per the PR description this checks out 👍🏾 g2g 🚀

farmerpaul committed:
fix: disable hidden activities in Assign Activity

Last-minute added to Figma here:
https://www.figma.com/design/6BNx63peNVzrHAF1FoZ78x/Multi-Informant-2024?node-id=13356-96153&t=rvs6Nveshvx07ymU-4
